python/tank/bootstrap/resolver.py

```diff
@@ -275,7 +274,133 @@ def find_matching_pipeline_configurations(self, pipeline_config_name, current_lo
         for pc in pipeline_configs:
             # make sure configuration matches our plugin id
             if self._match_plugin_id(pc.get("plugin_ids")) or self._match_plugin_id(pc.get("sg_plugin_ids")):
-                yield pc
+                path = ShotgunPath.from_shotgun_dict(pc)
+                # If a location was specified to get access to that pipeline, return it. Note that we are
+                # potentially returning pipeline configurations that have been configured for one platform but
+                # not all.
+                if pc.get("descriptor") or pc.get("sg_descriptor") or path:
+                    yield pc
+                else:
+                    log.warning("Pipeline configuration is missing a 'path' or 'descriptor' field: %s" % pc)
+
+    def _keep_latest_primary(self, configs, level_name):
+        """
+        Keeps the first primary pipeline configuration and logs warnings if where are any extra ones.
+
+        :param list configs: Pipeline configurations entities sorted by id from lowest to highest.
+        :param str level_name: Name of the scope for the pipeline configurations.
+
+        :returns: The first pipeline configuration from ``configs`` or ``None`` if it was empty.
+        """
+        first, remainder = configs[0: 1], configs[1:]
+
+        if remainder:
+            log.warning(
+                "Too many %s level pipeline configurations found. The following were skipped",
```

Comment:
We first warn the user if there are too many project level primaries, then we warn if there are two many site primaries.

Then we discard the site one if there is a project one. That doesn't generate a warning, but does log something at the debug level.